Feat/24 more verbose analyze process

rshayer95 opened the pull request:
Make the analyze process more verbose or add loading in CLI [#24](https://github.com/Repello-AI/Agent-Wiz/issues/24)

**Enhancements:**

- **Added Timer for Process Execution:** The analyzer CLI command now includes a timer that shows the elapsed time for task completion. This provides real-time feedback about the execution duration.

-  **Improved User Feedback:** The timer updates continuously on the same line during execution, reducing console clutter and offering a smoother experience. In case of errors or interruptions, the timer may appear on different lines temporarily.

- **Removed AgentChat_graph.json:** This file, which was used solely for testing purposes, has been removed from the repository as it is no longer needed.

**Description:**
- The timer feature enhances user experience by visually displaying the time it takes for the analyzer command to execute, allowing for more efficient tracking of task performance.
- The AgentChat_graph.json file has been removed, as it was only required for testing and is no longer relevant to the project.

Commit:
feat: Added Timer for CLI Analyzer so that it does not stay idle while generating report

Commit:
feat: Added Timer for CLI Analyzer so that it does not stay idle while generating report

Commit:
Removed AgentChat_graph that I used for testing the command.

Commit:
Merge branch 'feat/24-more-verbose-analyze-process' of https://github.com/rshayer95/Agent-Wiz into feat/24-more-verbose-analyze-process

rshayer95 committed:
Delete agentchat_graph.json

rshayer95 reviewed (commented):
Removed Testing JSON input file.

Aaditya-G commented on .env.sample:
```diff
@@ -1 +1 @@
-OPENAI_API_KEY=your_openai_api_key
+OPENAI_API_KEY=you_api_key
```
this change is not necessary

rshayer95 replied:
reverted to original

Aaditya-G commented on src/repello_agent_wiz/cli.py:
```diff
@@ -1,9 +1,44 @@
+            sys.stdout.write(f"\r⏳ {label}... Elapsed time: {elapsed:.1f}s")
+            sys.stdout.write("\n")
+            sys.stdout.flush()
+            time.sleep(0.1)
```
you can use tqdm or any another standard library, this output is not desirable,it looks like a spam of messages
<img width="788" height="413" alt="image" src="https://github.com/user-attachments/assets/673a2f38-ff17-4849-9730-e109046e360b" />

rshayer95 replied:
Fixed
<img width="285" height="36" alt="Screenshot 2025-10-14 at 1 10 15 PM" src="https://github.com/user-attachments/assets/e2a8435c-aec5-456a-8fe3-c2d1cfb9e51b" />

Aaditya-G commented on src/repello_agent_wiz/cli.py:
```diff
@@ -57,7 +92,8 @@ def main():
         case "analyze":
-            generate_maestro_analysis_report(args.input)
+            run_with_timer(generate_maestro_analysis_report, args.input, label="Generating threat model report")
+            # generate_maestro_analysis_report(args.input)
```
remove the comment

rshayer95 replied:
Comment Removed

Aaditya-G commented on src/repello_agent_wiz/cli.py:
```diff
@@ -1,9 +1,44 @@
+        *args, **kwargs: Arguments to pass to the function.
+    """
+    stop_event = threading.Event()
+    start_time = time.time()  # <-- define early so it's always available
```
remove any comments

rshayer95 replied:
Removed unnecessary comments

Aaditya-G reviewed (changes requested):
Kindly address the changes @rshayer95

rshayer95 commented:
Okay, will finalize the changes and submit back

rshayer95 committed:
fix: clean up timer output and revert unnecessary .env.sample change

- Fixed timer to display smoothly without spamming multiple lines
- Removed unnecessary comments for cleaner output
- Reverted .env.sample change to original placeholder value (OPENAI_API_KEY=your_openai_api_key)

rshayer95 commented:
Made the fixes. Please review and let me know if anything missed. Otherwise, we are good to go

Aaditya-G reviewed (changes requested):
@rshayer95  the timer in your current implementation is not visible in the cli, can you please use tqdm instead?

rshayer95 commented:
ok

rshayer95 committed:
Enhance: use tqdm for visible CLI progress in analyze process

Replaced custom timer with tqdm-based progress indicator to provide
visible feedback during the analyze process. This improves user experience
by displaying a dynamic progress bar and elapsed time in the CLI.

rshayer95 commented:
@Aaditya-G Added tqdm for the timer, please let me know if it is working as expected and if anything is missing. If everything's working as expected, we are good to go.

rshayer95 committed:
Merge branch 'master' into feat/24-more-verbose-analyze-process
